engine/classes/Elgg/Http/Request.php

```diff
@@ -35,424 +38,32 @@
-	 *
-	 * @return string
-	 */
-	public function getHttpHost() {
-		$scheme = $this->getScheme();
-		$port = $this->getPort();
-
-		if (('http' == $scheme && $port == 80) || ('https' == $scheme && $port == 443)) {
-			return $this->getHost();
-		}
-
-		return $this->getHost() . ':' . $port;
-	}
-
-	/**
-	 * Returns the host name.
-	 *
-	 * @return string
-	 *
-	 * @throws \UnexpectedValueException when the host name is invalid
-	 */
-	public function getHost() {
-		if (!$host = $this->headers->get('HOST')) {
-			if (!$host = $this->server->get('SERVER_NAME')) {
-				$host = $this->server->get('SERVER_ADDR', '');
-			}
-		}
-
-		// trim and remove port number from host
-		// host is lowercase as per RFC 952/2181
-		$host = strtolower(preg_replace('/:\d+$/', '', trim($host)));
-
-		// as the host can come from the user (HTTP_HOST and depending on the configuration,
-		// SERVER_NAME too can come from the user)
-		// check that it does not contain forbidden characters (see RFC 952 and RFC 2181)
-		if ($host && !preg_match('/^\[?(?:[a-zA-Z0-9-:\]_]+\.?)+$/', $host)) {
-			throw new \UnexpectedValueException('Invalid Host');
-		}
-
-		return $host;
-	}
-
-	/**
-	 * Is this request using SSL
-	 *
-	 * @return bool
-	 */
-	public function isSecure() {
-		return 'on' == strtolower($this->server->get('HTTPS')) || 1 == $this->server->get('HTTPS');
-	}
-
-	/**
-	 * Returns the port on which the request is made.
-	 *
-	 * @return string
-	 */
-	public function getPort() {
-		return $this->server->get('SERVER_PORT');
-	}
-
-	/**
-	 * Returns the root url from which this request is executed.
-	 *
-	 * The base URL never ends with a /. If the URL directly calls the
-	 * executed script (e.g. index.php), this is included.
-	 *
-	 * Suppose Elgg is installed at http://localhost/mysite/ :
-	 *
-	 * * http://localhost/mysite/foo returns '/mysite'
-	 * * http://localhost/mysite/index.php/foo returns '/mysite/index.php'
-	 *
-	 * @return string The raw url (i.e. not urldecoded)
-	 */
-	public function getBaseUrl() {
-		if (null === $this->baseUrl) {
-			$this->baseUrl = $this->prepareBaseUrl();
-		}
-
-		return $this->baseUrl;
-	}
-
-	/**
-	 * Returns the path being requested relative to the executed script.
-	 *
-	 * The path info always starts with a /, does not contain the query string,
-	 * and does not include the path to the Elgg installation.
-	 *
-	 * Suppose Elgg is installed at http://localhost/mysite/ :
-	 *
-	 * * http://localhost/mysite/ returns an empty string
-	 * * http://localhost/mysite/about returns '/about'
-	 * * http://localhost/mysite/enco%20ded returns '/enco%20ded'
-	 * * http://localhost/mysite/about?var=1 returns '/about'
-	 *
-	 * @return string The raw path (i.e. not urldecoded)
-	 */
-	public function getPathInfo() {
-		if (null === $this->pathInfo) {
-			$this->pathInfo = $this->preparePathInfo();
-		}
-
-		return $this->pathInfo;
-	}
+class Request extends SymfonyRequest {
```

Comment:
Definitely prefer using symfony for implementation but not tying ourselves to it (just in case...).

For the caching api I defined an interface we wanted to support (Pool), then implemented that interface with the class `StashPool` and composed Stash's pool implementation. I'd recommend doing the same here.